### DISCUSSION TITLE
How to instantly trigger Subscriptions

### DISCUSSION THREAD

**Discussion by jimmysafe** [MichalLytek/type-graphql#846] [Q&A]
Hello Everyone, hopefully someone can help me out on this.

I have implemented a subscription server for my app. Everything works fine and Subscritpion are successfully Listening for changes and return correct data.

Unfortunately i am not able to retrieve data instantly when i trigger a subscription (in the playground or in useSubscription hook in front end) and i have a constantly loading state that ends only when a subscription is triggered by a mutation or query.

What i am trying to achieve is to retrieve data whenever the subscription starts (in useSubscription or when i press the play button in the playground)

this is  my code and there is also a screenshot of the playground.

```
@Resolver()
export class AuctionResolver {

	@Subscription(() => Auction, {
		topics: ({ args }) => args.topic
	})
	async auction(@Arg('topic') topic: string): Promise<Auction | null> {
		const auction = await AuctionModel.findOne({ name: topic }).populate(auctionPopulate);
		if (!auction) throw new ApolloError('Auction not found.');
		return auction;
	}

	@Query(() => Auction)
	async singleAuction(
		@Arg('auctionName') auctionName: string,
		@PubSub() pubsub: PubSubEngine
	): Promise<Auction | null> {
		const auction = await AuctionModel.findOne({ name: auctionName }).populate(auctionPopulate);
		if (!auction) throw new ApolloError('Auction not found');

		await pubsub.publish(auction.name, { auction });

		return auction;
	}
}
```

![image](https://user-images.githubusercontent.com/12184812/113510481-537a0400-955b-11eb-874a-545387e909db.png)

as you can see in the playground the Subscription is correctly listening for changes, however i would need this to display instant data when i click the play button.

I had achieved this without Type-graphql but i cant seem to make it work in here.

Thanks in advance,

**Comment by MichalLytek** [comment]
Subscriptions are intended to use as a notification about some change in the system, like new e-mail in your inbox.
So it's working as intended - you can get all past emails by using a query and use subscription to instantly receive new emails.

In case of "display instant data when i click the play button" in this example context, I'm not sure what data you would expect to see after subscribing - there's not new emails yet, so it's in a loading state.

> I had achieved this without Type-graphql but i cant seem to make it work in here.

So please show your standard `graphql-js` code. In TypeGraphQL you should be able to achieve the same, there's `subscribe` option in the decorator to provide your own pub-sub implementation or async iterator.
